### ISSUE TITLE
LimitPropertyVisibilityRule that allows to specify visibility of properties via meta information

### ISSUE THREAD

**Issue by vanniktech** [arturbosch/detekt#208]
I have this crazy idea that I'd like to constraint certain properties to a specific visibility modifier since in Kotlin everything is public by default.

For now flagging the following code

```
@BindView(R2.id.activityAboutToolbar) lateinit var toolbar: Toolbar
```

would be enough, since I want the property to be internal.

I think that as a first step declaring the annotations is enough so something like:

```yaml
LimitPropertyVisibilityRule:
  1:
    package: "butterknife.*"
    visibilityModifier: "internal"
```

What do you guys think?

**Comment by arturbosch** [comment]
The rule reminds you that this property could be internal? Interesting, should this rule only work on annotations?

**Comment by vanniktech** [comment]
For now, I'd assume so. It's mostly for properties that need to be accessible from within the same package due to Class generation where I want to limit the visibility (e.g. in the case of ButterKnife where they generate a new class within the same package and in there set the value of `toolbar`).

**Comment by vanniktech** [comment]
This would only make sense with some quickfixes (#710) as you don't want this to get flagged and then correct every line by hand. Hence I'm closing it.
